[Merged by Bors] - feat(Positivity): add positivity extensions for `NNReal.rpow` and `ENNReal.rpow`

HugLycan opened the pull request:
---
This PR adds positivity extensions for `NNReal.rpow` and `ENNReal.rpow`. 
related issue #6038
@fpvandoorn 

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

HugLycan committed:
Update NNReal.lean

feat: add `positivity` extension for `NNReal.rpow` & `ENNReal.rpow`

HugLycan committed:
feat: add `positivity` extension for `NNReal.rpow` & `ENNReal.rpow`

grunweg commented:
Thanks for your PR. Making positivity more powerful is always great!
Can you add a few tests to show this works? `MathlibTest/positivity.lean` is a reasonable place to put them.

HugLycan committed:
Update positivity.lean

add some examples

HugLycan committed:
Update NNReal.lean

avoid simp by using particular tactics and theorems

HugLycan committed:
Merge branch 'leanprover-community:master' into feat/positivity-nnreal

fpvandoorn reviewed (commented):
This looks good! Here are a few comments.

fpvandoorn commented:
PS: once you have addressed all the comments, you can write a comment `-awaiting-author` to remove the corresponding label.

HugLycan committed:
Apply review suggestions

- remove `nonneg`
- merge `evalNNRealRpowZero` into `evalNNRealRpow`
- merge `evalENNRealRpowZero` into `evalENNRealRpow`
- use `zero_le` instead of `NNReal.coe_nonneg`
- add the case for positive finite base in `evalENNRealRpow`
- temperarily put the aux function `isFiniteM?` in this file
- add two more tests

HugLycan committed:
Merge branch 'master' into feat/positivity-nnreal

HugLycan committed:
Merge branch 'master' into feat/positivity-nnreal

HugLycan commented:
-awaiting-author
